### PR TITLE
Charm & terraform plan for hardware-api

### PR DESCRIPTION
**Pull request by mz2**
A first stab at a charm and a terraform plan for hardware-api. GitHub automation for publishing the image and charm in a follow-up PR.

### CODE REVIEW

**Comment by nadzyah** [comment]
Seems that you have to rebase the branch :grimacing:

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: hwctl
+base: core22
+version: '0.1' #
```
Comment (nadzyah) [inline snap/snapcraft.yaml:3]: Shouldn’t it match the project version?

**Review by nadzyah — CHANGES_REQUESTED** [review]
Happy to see these changes! Most comments are related to probably incorrectly rebasing from main, so I’ve added the comments to make sure the documentation is correct

File: README.md [hunk 2]
```diff
@@ -2,13 +2,15 @@
 
 The repo contains the API server and client for retrieving hardware information.
 
-
 ## Prerequisites for Deploying Locally
 
+- `poetry` vesion 1.6.1 or later for server deployment
+  (https://python-poetry.org/docs/)
```
[the remark this answers is not in the document]
Comment (mz2) [reply README.md:8]: Ah, that was my bad, will fix.

File: snap/snapcraft.yaml (hunk 1 above)
Comment (mz2) [reply snap/snapcraft.yaml:3]: Best to drive all places where we refer to the version from git tags.

File: README.md [hunk 3]
```diff
@@ -2,56 +2,66 @@
...
 
-
 ## Prerequisites for Deploying Locally
 
-* Install [docker](https://docs.docker.com/engine/install/ubuntu/) and [setup permissions](https://docs.docker.com/engine/install/linux-postinstall/) for server deployment
-* Install `rust` and `cargo` for client deployment (https://www.rust-lang.org/tools/install)
-
+- Install [docker](https://docs.docker.com/engine/install/ubuntu/) and
+  [setup permissions](https://docs.docker.com/engine/install/linux-postinstall/)
+  for server deployment <<<<<<< HEAD
```
Comment (nadzyah) [inline README.md:9]: Probably you forgot to delete it after rebasing

File: README.md [hunk 4]
```diff
@@ -2,56 +2,66 @@
...
 
 ## Running API Server With Docker
 
-Go to the `server/` directory in the project. The following command builds the hwapi server:
+Go to the `server/` directory in the project. The following command builds the
+hwapi server:
 
 ```bash
-:server$ docker-compose up --attach-dependencies --build hwapi-dev
+docker-compose up --attach-dependencies --build hwapi-dev
```
Comment (nadzyah) [inline README.md:19]: I’d like to keep the directory indicator to specify where to run this command

File: README.md [hunk 5]
```diff
@@ -2,56 +2,66 @@
...
 ```
 
-If you don't have any db initialised (in the `server/data/hwapi.db` location), the command above will also initialise a DB with sample data.
+If you don't have any db initialised (in the `server/data/hwapi.db` location),
```
Comment (nadzyah) [inline README.md:22]:
```suggestion
If you don't have any db initialised (in the `server/data/hwapi-dev.db` location),
```

File: README.md [hunk 6]
```diff
@@ -2,56 +2,66 @@
...
-
 ## Build and Run the Reference CLI Tool (`hwctl`)
 
 ```bash
-$ export CERTIFICATION_STATUS=2
-$ cd client/hwctl
-:client/hwlib$ cargo run
+export CERTIFICATION_STATUS=2
+cd client/hwctl
+cargo run
```
Comment (nadzyah) [inline README.md:64]: Still would like to keep the folder indicators

File: README.md (hunk 3 above)
Comment (mz2) [reply README.md]: Oops, fixed

File: README.md (hunk 4 above)
Comment (mz2) [reply README.md:19]: But but then I can't copy and paste from the README?

File: README.md (hunk 6 above)
Comment (mz2) [reply README.md:64]: But then I can't copy and paste from the README :-(

**Pull request merged**